File: 프로그래머스/2/42586. 기능개발/기능개발.kt
class Solution {
    fun sum(progresses: IntArray, speeds:IntArray)
    {
        repeat(progresses.size) {progresses[it] += speeds[it]}
    }

    fun releaseCount(startIndex: Int, progresses: IntArray): Int {
        var release = 1
        for (j in (startIndex + 1).until(progresses.size)) {
            if (progresses.get(j) >= 100) release++
            else break
        }
        return release
    }

    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        var answer = intArrayOf()

        var i = 0
        while (i < progresses.size) {
            sum(progresses, speeds)
            if (progresses[i] >= 100) {
                val release: Int = releaseCount(i, progresses)
                answer+=release
                i += release - 1
            } else {
                i--
            }
            i++
        }

        return answer
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

void summit(vector<int>& progresses, vector<int>& speeds)
{
    for (int i = 0; i < progresses.size(); i++)
    {
        progresses[i] += speeds[i];
    }
}

int release_Count(int start_index, vector<int> progresses, vector<int> speeds)
{
    int release = 1;
    for (int j = start_index + 1; j < progresses.size(); j++)
    {
        if (progresses[j] >= 100)
            release++;
        else
            break;
    }
    return release;
}

vector<int> solution(vector<int> progresses, vector<int> speeds)
{
    vector<int> answer;

    for (int i = 0; i < progresses.size(); i++)
    {
        summit(progresses, speeds);
        if (progresses[i] >= 100)
        {
            int release=release_Count(i, progresses, speeds);
            answer.push_back(release);
            i+=release-1;
        }
        else
        {
            i--;
        }
    }
   
    return answer;
}